leader-followers/src/main/java/com/iluwatar/leaderfollower/App.java

```diff
@@ -0,0 +1,106 @@
+ * <p>
+ * • It can minimize priority inversion because no extra queueing is introduced in the server. When combined with
+ * real-time I/O subsystems, the Leader/Followers thread pool model can reduce sources of non-determinism in server
+ * request processing significantly.
+ * <p>
+ * • It does not require a context switch to handle each event, reducing the event dispatching latency. Note that
+ * promoting a follower thread to fulfill the leader role does require a context switch. If two events arrive
+ * simultaneously this increases the dispatching latency for the second event, but the performance is no worse than
+ * Half-Sync/Half-Reactive thread pool implementations.
+ * <p>
+ * 2)Programming simplicity. The Leader/Follower pattern simplifies the programming of concurrency models where multiple
+ * threads can receive requests, process responses, and demultiplex connections using a shared handle set.
+ * <p>
+ * However, the Leader/Followers pattern has the following liabilities:
+ * <p>
+ * 1)Implementation complexity. The advanced variants of the Leader/Followers pattern are harder to implement than
+ * Half-Sync/Half-Reactive thread pools. In particular, when used as a multi-threaded connection multiplexer,
+ * the Leader/Followers pattern must maintain a pool of follower threads waiting to process requests. This set must
+ * be updated when a follower thread is promoted to a leader and when a thread rejoins the pool of follower threads.
+ * All these operations can happen concurrently, in an unpredictable order. Thus, the Leader/Follower pattern
+ * implementation must be efficient, while ensuring operation atomicity.
+ * <p>
+ * 2)Lack of flexibility. Thread pool models based on the Half-Sync/Half-Reactive variant of the Half-Sync/Half-Async
+ * pattern allow events in the queueing layer to be discarded or re-prioritized. Similarly, the system can maintain
+ * multiple separate queues serviced by threads at different priorities to reduce contention and priority inversion
+ * between events at different priorities. In the Leader/Followers model, however, it is harder to discard or reorder
+ * events because there is no explicit queue. One way to provide this functionality is to offer different levels of
+ * service by using multiple Leader/Followers groups in the application, each one serviced by threads at different
+ * priorities.
+ * <p>
+ * 3)Network I/O bottlenecks. The Leader/Followers pattern, as described in the Implementation section, serializes
+ * processing by allowing only a single thread at a time to wait on the handle set. In some environments, this design
+ * could become a bottleneck because only one thread at a time can demultiplex I/O events. In practice, however, this
+ * may not be a problem because most of the I/O-intensive processing is performed by the operating system kernel. Thus,
+ * application-level I/O operations can be performed rapidly.
+ */
```

Comment:
The pattern is explained in very detailed level. Maybe the text could be shortened a bit? Also it would be nice to explain here how the example implements the pattern.